ethToElrond/bridgeExecutors/signaturesHolder_test.go

```diff
@@ -0,0 +1,272 @@
+		}
+
+		sh := newSignatureHolder()
+		sh.ProcessNewMessage(msg, ethMsg)
+		sh.ProcessNewMessage(msg2, ethMsg2)
+		compareEthSignatureMessageLists(t, []*core.EthereumSignature{ethMsg, ethMsg2}, sh.ethMessages)
+		compareSignedMessageLists(t, []*core.SignedMessage{msg, msg2}, sh.AllStoredSignatures())
+	})
+}
+
+func TestSignatureHolder_Signatures(t *testing.T) {
+	t.Parallel()
+
+	t.Run("unique signatures should work", func(t *testing.T) {
+		msg := &core.SignedMessage{
+			Payload:        []byte("payload"),
+			Signature:      []byte("sig"),
+			PublicKeyBytes: []byte("pk"),
+			Nonce:          1,
+		}
+		ethMsg := &core.EthereumSignature{
+			Signature:   []byte("eth sig"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		msg2 := &core.SignedMessage{
+			Payload:        []byte("payload2"),
+			Signature:      []byte("sig2"),
+			PublicKeyBytes: []byte("pk2"),
+			Nonce:          1,
+		}
+		ethMsg2 := &core.EthereumSignature{
+			Signature:   []byte("eth sig 2"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		sh := newSignatureHolder()
+		sh.ProcessNewMessage(msg, ethMsg)
+		sh.ProcessNewMessage(msg2, ethMsg2)
+
+		compareBytesSlicesLists(t, [][]byte{[]byte("eth sig"), []byte("eth sig 2")}, sh.Signatures([]byte("eth msg")))
+
+		sh.clearStoredSignatures()
+
+		assert.Equal(t, 0, len(sh.Signatures([]byte("eth msg"))))
+	})
+	t.Run("same signatures should return the unique ones", func(t *testing.T) {
+		msg := &core.SignedMessage{
+			Payload:        []byte("payload"),
+			Signature:      []byte("sig"),
+			PublicKeyBytes: []byte("pk"),
+			Nonce:          1,
+		}
+		ethMsg := &core.EthereumSignature{
+			Signature:   []byte("eth sig"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		msg2 := &core.SignedMessage{
+			Payload:        []byte("payload2"),
+			Signature:      []byte("sig2"),
+			PublicKeyBytes: []byte("pk2"),
+			Nonce:          1,
+		}
+		ethMsg2 := &core.EthereumSignature{
+			Signature:   []byte("eth sig 2"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		msg3 := &core.SignedMessage{
+			Payload:        []byte("payload3"),
+			Signature:      []byte("sig3"),
+			PublicKeyBytes: []byte("pk3"),
+			Nonce:          1,
+		}
+		ethMsg3 := &core.EthereumSignature{
+			Signature:   []byte("eth sig 2"),
+			MessageHash: []byte("eth msg"),
+		}
+
+		sh := newSignatureHolder()
+		sh.ProcessNewMessage(msg, ethMsg)
+		sh.ProcessNewMessage(msg2, ethMsg2)
+		sh.ProcessNewMessage(msg3, ethMsg3)
+
+		compareBytesSlicesLists(t, [][]byte{[]byte("eth sig"), []byte("eth sig 2")}, sh.Signatures([]byte("eth msg")))
+	})
+	t.Run("same signatures should return filter by message", func(t *testing.T) {
+		msg := &core.SignedMessage{
+			Payload:        []byte("payload"),
+			Signature:      []byte("sig"),
+			PublicKeyBytes: []byte("pk"),
+			Nonce:          1,
+		}
+		ethMsg := &core.EthereumSignature{
+			Signature:   []byte("eth sig"),
+			MessageHash: []byte("eth msg 1"),
+		}
+
+		msg2 := &core.SignedMessage{
+			Payload:        []byte("payload2"),
```

Comment:
create a function getSignedMessages(num int) -> that is returning num generated messages

Reply:
refactored